File: apprefurbished/src/main/java/com/androidcollider/easyfin/managers/format/date/DateFormatManager.java
package com.androidcollider.easyfin.managers.format.date;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * @author Ihor Bilous
 */

public class DateFormatManager {

    public static final String DAY_MONTH_YEAR_DOTS = "dd.MM.yyyy";
    public static final String DAY_MONTH_YEAR_SPACED = "dd MMMM yyyy";

    public String dateToString(Date date, String dateFormat) {
        return new SimpleDateFormat(dateFormat, Locale.getDefault()).format(date);
    }

    public Date stringToDate(String dateStr, String dateFormat) {
        Date date = null;
        try {
            date = new SimpleDateFormat(dateFormat, Locale.getDefault()).parse(dateStr);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return date;
    }

    public String longToDateString(long dateLong, String dateFormat) {
        return new SimpleDateFormat(dateFormat, Locale.getDefault()).format(new Date(dateLong));
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/format/date/DateFormatManager.kt
package com.androidcollider.easyfin.common.managers.format.date

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

/**
 * @author Ihor Bilous
 */
class DateFormatManager {

    fun dateToString(date: Date, dateFormat: String): String {
        return SimpleDateFormat(dateFormat, Locale.getDefault()).format(date)
    }

    fun stringToDate(dateStr: String, dateFormat: String): Date? {
        var date: Date? = null
        try {
            date = SimpleDateFormat(dateFormat, Locale.getDefault()).parse(dateStr)
        } catch (e: ParseException) {
            e.printStackTrace()
        }
        return date
    }

    fun longToDateString(dateLong: Long, dateFormat: String): String {
        return SimpleDateFormat(dateFormat, Locale.getDefault()).format(Date(dateLong))
    }

    companion object {
        const val DAY_MONTH_YEAR_DOTS = "dd.MM.yyyy"
        const val DAY_MONTH_YEAR_SPACED = "dd MMMM yyyy"
    }
}